### PR TITLE
Misc CVM fixes for AWS

### PR DESCRIPTION
**Pull request by bpradipt**
1. Add uefi support when registering AMI created from raw image. This is tested with mkosi fedora image
2. Adjust disable-cvm option to align with other providers

### CODE REVIEW

File: pkg/adaptor/cloud/aws/manager.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (_ *Manager) ParseCmd(flags *flag.FlagSet) {
 	flags.IntVar(&awscfg.RootVolumeSize, "root-volume-size", 30, "Root volume size (in GiB) for the Pod VMs")
 	// Setting disable-cvm to true as there are still some rough edges with CVMs.
```
Comment (katexochen) [inline pkg/adaptor/cloud/aws/manager.go:37]: I guess we should also remove the comment?

Comment (bpradipt) [reply pkg/adaptor/cloud/aws/manager.go:37]: Ah yes. Thanks for spotting it

**Review by snir911 — APPROVED** [review]

**Review by katexochen — APPROVED** [review]
Code lgtm, but didn't test as I currently don't have a cluster available.

**Comment by bpradipt** [comment]
I tested it with a fedora image generated using mkosi and instance type of m6a.large

The following image shows that AMD SEV-SNP is enabled
image

The following image shows the SNP tags automatically added by AWS
image

What is still pending, is to get vtpm based attestation working. 
@katexochen do you have any idea from your experience with constellation ?

**Comment by katexochen** [comment]
> What is still pending, is to get vtpm based attestation working. @katexochen do you have any idea from your experience with constellation ?

Did you use uplosi for the image upload? The image should be tagged in the right way to enable vTPM: https://github.com/edgelesssys/uplosi/blob/main/aws/uploader.go#L376

Also maybe we need to use boot mode "uefi" instead of "uefi-preferred".

You can check if an AMI has vTPM enabled by [following this AWS doc](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-nitrotpm-support-on-ami.html).

**Comment by bpradipt** [comment]
> https://github.com/edgelesssys/uplosi/blob/main/aws/uploader.go#L376

I didn't use uplosi for AWS. I'll try and let you know.

**Comment by bpradipt** [comment]
Changed from `uefi-preferred` to `uefi` in the image upload script.

**Pull request merged**